gp-includes/assets-loader.php

```diff
@@ -1,84 +1,129 @@
+add_action( 'init', 'gp_register_scripts' );
 
 /**
- * Here we abstract WordPress core's enqueuing functions because...
- * 1. We don't want to print scripts and styles that are meant for the WordPress theme
- * 2. GlotPress enqueues scripts and styles from its template files and if we do that
- *    with wp_enqueue_script() and wp_enqueue_style() WordPress complains that
- *    those functions should only be called inside of wp_enqueue_scripts()
+ * Enqueue one or more styles.
+ *
+ * @since 2.2.0
+ *
+ * @param string|array $handles A single style handle to enqueue or an array or style handles to enqueue.
  */
-
-function gp_enqueue_scripts() {
-	global $gp_enqueued_styles, $gp_enqueued_scripts;
-
-	if ( ! empty( $gp_enqueued_scripts ) ) {
-		foreach ( $gp_enqueued_scripts as $handle ) {
-			wp_enqueue_script( $handle );
-		}
-	}
-
-	if ( ! empty( $gp_enqueued_styles ) ) {
-		foreach ( $gp_enqueued_styles as $handle ) {
+function gp_enqueue_styles( $handles ) {
+	// Make sure $handles is an array to simplify the next loop.
+	$handles = (array) $handles;
+
+	// Loop through each handle we've been asked to enqueue.
+	foreach ( $handles as $handle ) {
+		// If we've already enqueued the style, we don't need to do it again.
+		if( ! in_array( $handle, GP::$styles ) ) {
+			// Store the handle name in the global array.
+			GP::$styles[] = $handle;
+
+			// Actually enqueue the handle via WordPress.
 			wp_enqueue_style( $handle );
 		}
 	}
 }
 
-add_action( 'wp_enqueue_scripts', 'gp_enqueue_scripts' );
+/**
+ * Enqueue one or more styles.
+ *
+ * This function is a wrapper for backwards compatibility.
+ *
+ * @since 1.0.0
+ *
+ * @param string|array $handles A single style handle to enqueue or an array or style handles to enqueue.
+ */
+function gp_enqueue_style( $handles ) {
+	gp_enqueue_styles( $handles );
+}
 
-function gp_enqueue_style( $handle ) {
-	global $gp_enqueued_styles;
+/**
+ * Enqueue one or more scripts.
+ *
+ * @since 2.2.0
+ *
+ * @param string|array $handles A single script handle to enqueue or an array of enqueue handles to enqueue.
+ */
+function gp_enqueue_scripts( $handles ) {
+	// Make sure $handles is an array to simplify the next loop.
+	$handles = (array) $handles;
 
-	$gp_enqueued_styles[] = $handle;
-}
+	// Loop through each handle we've been asked to enqueue.
+	foreach ( $handles as $handle ) {
+		// Store the handle name in the global array.
+		GP::$scripts[] = $handle;
 
-function gp_enqueue_script( $handle ) {
-	global $gp_enqueued_scripts;
+		// Actually enqueue the handle via WordPress.
+		wp_enqueue_script( $handle );
+	}
+}
 
-	$gp_enqueued_scripts[] = $handle;
+/**
+ * Enqueue one or more scripts.
+ *
+ * This function is a wrapper for backwards compatibility.
+ *
+ * @since 1.0.0
+ *
+ * @param string|array $handles A single script handle to enqueue or an array of enqueue handles to enqueue.
+ */
+function gp_enqueue_script( $handles ) {
+	gp_enqueue_scripts( $handles );
 }
```

Comment:
We shouldn't provide two functions which do the same. `gp_enqueue_script()` should continue to only support one handle.

```php
function gp_enqueue_script( $handle ) {
    gp_enqueue_scripts( array( $handle ) );
}
```

Same for `gp_enqueue_style()`.